S4TF_Tutorial_2.ipynb

```diff
@@ -0,0 +1,875 @@
+        "    <a target=\"_blank\" href=\"Link to be updated\"><img src=\"https://www.tensorflow.org/images/GitHub-Mark-32px.png\" />GitHub link to be updated accordingly</a>\n",
+        "  </td>\n",
+        "</table>"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "F8YVA_634OFk"
+      },
+      "source": [
+        "Welcome to this Colab where you will train your first Machine Learning model!\n",
+        "\n",
+        "We'll try to keep things simple here, and only introduce basic concepts. Later Colabs will cover more advanced problems.\n",
+        "\n",
+        "The problem we will solve is to convert from Celsius to Fahrenheit, where the approximate formula is:\n",
+        "\n",
+        "$$ f = c \\times 1.8 + 32 $$\n",
+        "\n",
+        "\n",
+        "Of course, it would be simple enough to create a conventional Python function that directly performs this calculation, but that wouldn't be machine learning.\n",
+        "\n",
+        "\n",
+        "Instead, we will give TensorFlow some sample Celsius values (0, 8, 15, 22, 38) and their corresponding Fahrenheit values (32, 46, 59, 72, 100).\n",
+        "Then, we will train a model that figures out the above formula through the training process."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "KaGgOzPQbGj3",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "import TensorFlow\n",
+        "\n",
+        "import Python\n",
+        "%include \"EnableIPythonDisplay.swift\"\n",
+        "IPythonDisplay.shell.enable_matplotlib(\"inline\")\n",
+        "let plt = Python.import(\"matplotlib.pyplot\")\n",
+        "\n",
+        "let np = Python.import(\"numpy\")  // Make numpy available using np."
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "wwJGmDrQ0EoB"
+      },
+      "source": [
+        "## Set up Training Data\n",
+        "\n",
+        "As we saw before, supervised Machine Learning is all about figuring out an algorithm given a set of inputs and outputs. Since the task in this Codelab is to create a model that can give the temperature in Fahrenheit when given the degrees in Celsius, we create two lists `celsiusTemperatures` and `fahrenheitTemperatures` that we can use to train our model."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "Q_XrcNGNbejX",
+        "colab_type": "code",
+        "outputId": "c8142a33-98ae-4952-e926-173c5c619fce",
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 136
+        }
+      },
+      "source": [
+        "let celsiusTemperatures: [Float] = [-40, -10,  0,  8, 15, 22,  38]\n",
+        "let input = Tensor<Float>(shape: TensorShape(7, 1), scalars: celsiusTemperatures)\n",
```

Comment:
[`TensorShape` conforms to `ExpressibleByArrayLiteral`](https://github.com/tensorflow/swift-apis/blob/c1b18f73be9d67c5b2be80c5dfd044791c0c1d2d/Sources/TensorFlow/Core/TensorShape.swift#L23). You can use `[7, 1]` directly.

```suggestion
        "let input = Tensor<Float>(shape: [7, 1], scalars: celsiusTemperatures)\n",
```